### PR TITLE
fix(liquidation-reporter): fixed broken data processing and update subgraph issues 

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

The liquidation reporter has been broken for about a month. This PR fixes it.

**Details**
There were a number of things that broke it, which have now been fixed. namely:

- we moved to a new graph URL that broken the position querying.
- we launched KPI options and call options using EMPS which the script did not know how to handle.
- there was a bug with stable coins. previously anything collateralized in DAI/USDC/USDT would not generate correctly.
- There was an issue with data structure modifiction that was not present until we had a bunch more expired contracts in the data set.

additional changes include:
- state handling refactors within data retrieval logic.
- modified excel report generator to be more robust against malformed data.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [X]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/reporters/liquidation-reporter/excel-writer.ts [hunk 1]
```diff
@@ -74,11 +74,13 @@ export function createExcelSheetFromLiquidationDrawDownData(drawDownData: any) {
     );
 
     collateralData.activeFinancialContracts.forEach((financialContractInfo: any, index: number) => {
-      sheet.cell(4 + index, 7).string(financialContractInfo.contractAddress);
```
Comment (chrismaree) [inline packages/reporters/liquidation-reporter/excel-writer.ts:77]: the changes here made the write more resilient to missing data.

File: packages/reporters/liquidation-reporter/index.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ fetchDataAndWriteToExcelFile()
       process.exit(0);
     }, 2000);
   })
-  .catch(() => {
+  .catch(e => {
+    console.error(e);
```
Comment (chrismaree) [inline packages/reporters/liquidation-reporter/index.ts:16]: easier debugging

File: packages/reporters/liquidation-reporter/liquidation-reporter.ts [hunk 3]
```diff
@@ -99,43 +97,34 @@ export async function fetchUmaEcosystemData() {
...
-        );
-        continue;
-      }
+      if (contractPositionsInfo.positions.length == 0) continue;
+
       // If we got to this point we know that the financial contract matches what was returned from the graph and
       // the contract has positions in it. We can now fetch some extra info about the contract.
       const financialContract = new web3.eth.Contract(getAbi("ExpiringMultiParty"), financialContractAddress);
-      const [collateralRequirementString, contractExpirationTime, contractPriceIdentifier] = await Promise.all([
+      const [collateralRequirementString, contractExpirationTime, contractPriceIdentifier] = await Promise.allSettled([
```
Comment (chrismaree) [inline packages/reporters/liquidation-reporter/liquidation-reporter.ts:110]: change to `allSettled` so some of the calls can fail. for example on the perp.

File: packages/reporters/liquidation-reporter/liquidation-reporter.ts [hunk 4]
```diff
@@ -99,43 +97,34 @@ export async function fetchUmaEcosystemData() {
...
       )[0];
       // If there is no position info, then continue. This would happen if there is data that is not on the graph.
-      if (contractPositionsInfo == undefined) {
-        continue;
-      }
+      if (contractPositionsInfo == undefined) continue;
 
       // If the contract has no positions, we should also continue. Remove the contract from the list of active contracts
-      if (contractPositionsInfo.positions.length == 0) {
-        uniqueCollateralTypes[collateralAddress].activeFinancialContracts = removeItemFromArrayOfObjects(
```
Comment (chrismaree) [inline packages/reporters/liquidation-reporter/liquidation-reporter.ts:108]: no longer need to use the `removeItemFromArrayOfObjects`. rather, we strip all empty elements later on.

File: packages/reporters/liquidation-reporter/liquidation-reporter.ts [hunk 5]
```diff
@@ -166,6 +155,17 @@ export async function fetchUmaEcosystemData() {
         if (samplePriceFeed.invertPrice) invertedPrice = true;
         if (samplePriceFeed.priceFeeds && samplePriceFeed.priceFeeds[0].invertPrice) invertedPrice = true;
 
+        // If the collateral type is stable, then we must invert the price feed by default. These kinds of feeds (such
+        // as the BasketSpreadPriceFeed) do not contain the `invertPrice` flag, but they are all "inverted". As additional
+        // stable collaterals are added this will need to be updated or the page will show up no liquidatable positions
+        // for that collateral type.
+        if (
```
Comment (chrismaree) [inline packages/reporters/liquidation-reporter/liquidation-reporter.ts:164]: this was a tricky one. all price feeds for stable collaterals are technically "inverted", such as `BasketSpreadPriceFeed`. Without this no stable coin liquidations show up correctly.

File: packages/reporters/liquidation-reporter/liquidation-reporter.ts [hunk 6]
```diff
@@ -233,32 +233,26 @@ export async function fetchUmaEcosystemData() {
         console.error(error);
       }
     }
-    // Finally, remove any elements that have undefined params.
```
Comment (chrismaree) [inline packages/reporters/liquidation-reporter/liquidation-reporter.ts:236]: previously we were editing an element while iterating over it. this is really hard to reason about and debug. I've changed it to a more expressive verbose syntax that makes it easier to see what's happening.

File: yarn.lock [hunk 7]
```diff
@@ -1363,7 +1363,7 @@
   dependencies:
```
Comment (chrismaree) [inline yarn.lock:1363]: diff from re-running `yarn`

**Review by nicholaspai — APPROVED** [review]
Thanks for fixing this!

**Review by daywiss — APPROVED** [review]
lgtm

**Pull request merged**